samples/server/petstore/rust-axum/output/petstore-with-fake-endpoints-models-for-testing/src/models.rs

```diff
@@ -2753,16 +2753,16 @@ pub struct FormatTest {
     pub int64: Option<i64>,
 
     #[serde(rename = "number")]
-    #[validate(range(min = 32.1, max = 543.2))]
+    #[validate(range(min = 32.1f64, max = 543.2f64))]
     pub number: f64,
 
     #[serde(rename = "float")]
-    #[validate(range(min = 54.3, max = 987.6))]
+    #[validate(range(min = 54.3f32, max = 987.6f32))]
     #[serde(skip_serializing_if = "Option::is_none")]
     pub float: Option<f32>,
 
     #[serde(rename = "double")]
-    #[validate(range(min = 67.8, max = 123.4))]
+    #[validate(range(min = 67.8f64, max = 123.4f64))]
```

Comment:
Within validate macros, `Type` is inferred implicitly. We don't need to explicit set it.

Reply:
Sadly, that is not true.
Take [this example declaration](https://gist.github.com/Victoria-Casasampere-BeeTheData/8fed2a93c9dbb97310da0a4027857a5b), the value specified in `maximum` is larger than i32, the default type of integer literals. Without this patch, the resulting code does not compile, as the type is not being inferred, and is defaulting to i32.

```
error: literal out of range for `i32`
  --> src/models.rs:21:34
   |
21 |             range(min = 0, max = 1231231232123),
   |                                  ^^^^^^^^^^^^^
   |
   = note: the literal `1231231232123` does not fit into the type `i32` whose range is `-2147483648..=2147483647`
   = help: consider using the type `i64` instead
   = note: `#[deny(overflowing_literals)]` on by default
```

Reply:
@Victoria-Casasampere-BeeTheData (cc @wing328 for your opinion)

IMHO, above compilation error is expected. Setting `1231231232123` is `malform` OpenAPI Specs to me. User should get fail compilation instead of bypassing.

Reply:
Can you provide a link to OpenAPI specification where it specifies, or clarifies, that this literal does not follow the spec?

From my research, this is a valid pattern on the spec and the resulting erroneous code is due to a bug in the generator. According to the schema object spec [1]:

> Unless stated otherwise, the keyword definitions follow those of JSON Schema and do not add any additional semantics [...]. Where JSON Schema indicates that behavior is defined by the application (e.g. for annotations), OAS also defers the definition of semantics to the application consuming the OpenAPI document.

The OpenAPI specification will follow the JSON Schema specification unless specified, but the application using the OpenAPI declaration (openapi-generator in this case) is free to define its own behaviour and the definition of semantics.

With this acknowledged, the data types spec [2] mentions which types are valid, this includes `integer`, which is the one used in the example declaration, but it also notes that `integer` is no different from `number`, as JSON (not schema) does not make a distinction between the two. `number` is defined as:

> An arbitrary-precision, base-10 decimal number value, from the JSON "number" value

In the JSON Schema [5]

Both `number` and `integer` do not have any default format specified, but the generator chooses to default to `double` and `int32` respectively (which follows spec).
The literal `1231231232123` fits within `int64`, which is a valid format defined by the OAS [2], not to mention it fits the JSON Schema definition [5] which does not have any bounds or sign specified, but even when `type: integer` or `type: number` with `format: int64` is specified, invalid code is generated that fails to compile, making this a bug.

Of course, the spec also mentions that `format` can be treated as a non-validating annotation, but in the rust-axum generator, this value is used to alter the integer type primitive that is used in the structure field, with even more fine grain control over the type when using the `minimum` and `maximum` validators (eg. `minimum: 0` leads to an unsigned integer type) (more on this later).

The OAS mentions that the validation follows JSON Schema Draft 2020-12 [3], this spec defines the validators `minimum` and `maximum` [4], and mentions that the value specified is a "number". Defining semantics, this mention of "number" could be the data type `number`, which would mean that any integer or floating point value with any sign and without bounds can be used as a literal in either field.

The JSON Schema spec also mentions that implementations may be altered to better support the specific languages [6], this would mean that even if all of what was mentioned previously wasn't specified, the implementation could still be altered to fix this bug, as in rust, all number primitives can be converted to the JSON format via serialization. 

To add to all of this, the custom handling that the rust-axum generator uses when the `minimum` and `maximum` validators are used to generate a more fitting primitive type, breaks when specifying a `maximum` bound that's above `i32::MAX`. The custom handling is done in [AbstractRustCodegen.java](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractRustCodegen.java#L102). This patch makes this feature actually usable, making this an even more important bug to fix.

Sources:
1. [OAS 3.1.1: Schema Object](https://spec.openapis.org/oas/v3.1.1.html#schema-object)
2. [OAS 3.1.1: Data Types](https://spec.openapis.org/oas/v3.1.1.html#data-types)
3. [OAS 3.1.1: Validation](https://spec.openapis.org/oas/v3.1.1.html#extended-validation-with-annotations)
4. [JSON Schema Validation: 6.2 Numeric validation](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#section-6.2)
5. [JSON Schema: 4.2.1 Instance data model](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-4.2.1)
6. [JSON Schema: 12.1 Format](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-00#section-12.1)

Reply:
@Victoria-Casasampere-BeeTheData 

You are right:
- I misunderstanding that `1231231232123` is out of int64 range.
- The problem here is handling `literal` in the `validator` crate's macros. 

Just try by myself:
```rust
#[derive(Debug, Validate)]
struct Data {
    #[validate(range(min = 18, max = 1231231232123))]
    data: i64,
}
```

Then the PR LGTM. Kindly ping @wing328 to merge this!

Reply:
And truly appreciate @Victoria-Casasampere-BeeTheData for great discussion and make a fix for Rust-Axum